plugins/inputs/statsd/running_stats.go

```diff
@@ -69,12 +77,32 @@ func (rs *RunningStats) AddValue(v float64) {
 		// Reached limit, choose random index to overwrite in the percentile array
 		rs.perc[rand.Intn(len(rs.perc))] = v
 	}
+
+	rs.MedLen = len(rs.med)
+	// Need to sort for median
+	sort.Float64s(rs.med)
+	if rs.MedLen < rs.MedLimit {
+		rs.med = append(rs.med, v)
+	} else {
+		// Reached limit, choose random index to overwrite in the median array
+		rs.med[rand.Intn(rs.MedLen)] = v
+	}
```

Comment:
Why not doing a FIFO here instead of randomness?

Reply:
yeah switched to fifo. might it cause a memory leak?

Reply:
Well that's why you should not sort the list here but keep the temporal order... :-)

Reply:
ah good point thank you!